ledger/src/blockstore.rs

```diff
@@ -2152,7 +2163,7 @@ impl Blockstore {
                 highest_primary_index_slot = Some(meta.max_slot);
             }
         }
-        if highest_primary_index_slot.is_some() {
+        if highest_primary_index_slot.is_some_and(|slot| slot != 0) {
```

Comment:
This feels a little dirty but couldn't think of a better way to do it.

Reply:
Yeah, really my fault for all the `clean_slot_0` stuff, though. This wfm!